lib/client.js

```diff
@@ -154,22 +220,29 @@ module.exports = function (dependencies) {
         resolve({ device, error });
       });
 
+      if (logger.enabled) {
```

Comment:
This block is wrong and should be removed - I believe this leads to a memory leak. The event handler for session were already handled elsewhere

A given session can have an arbitrary number of requests.

This is the handler for a request - so this ends up registering an arbitrary number of callbacks for frameError

Reply:
This seems to definitely be the case. We were seeing memory usage increasing linearly in our staging environment, but after removing this block memory is stable again

Reply:
So far, this seems to be working reasonably well in production, using the suggested patches as https://github.com/TysonAndre/node-apn/commits/ae8aa835b1689604047b78bee567bdfe706ff197 (with DEBUG=apn)

Reply:
Fixed. it should be `request` and not `this.session`